File: infrastructure/src/main/java/com/study/platform/domain/post/infrastructure/StudyPostRepositoryAdapter.java
package com.study.platform.domain.post.infrastructure;

import com.study.platform.domain.post.model.StudyPost;
import com.study.platform.domain.post.model.StudyPostRepository;
import com.study.platform.domain.post.model.StudyPostStatus;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Repository
@RequiredArgsConstructor
public class StudyPostRepositoryAdapter implements StudyPostRepository {

    private final StudyPostJpaRepository studyPostJpaRepository;

    @Override
    public StudyPost save(StudyPost post) {
        return studyPostJpaRepository.saveAndFlush(post);
    }

    @Override
    public List<StudyPost> saveAll(List<StudyPost> posts) {
        return studyPostJpaRepository.saveAll(posts);
    }

    @Override
    public void delete(StudyPost post) {
        studyPostJpaRepository.delete(post);
    }

    @Override
    public Optional<StudyPost> findById(UUID id) {
        return studyPostJpaRepository.findById(id);
    }

    @Override
    public Optional<StudyPost> findByIdWithAuthor(UUID postId) {
        return studyPostJpaRepository.findByIdWithAuthor(postId);
    }

    @Override
    public List<StudyPost> findExpiredPosts(LocalDateTime now, StudyPostStatus status) {
        return studyPostJpaRepository.findExpiredPosts(now, status);
    }

    @Override
    public List<StudyPost> findDeadlineReminderPosts(LocalDateTime start, LocalDateTime end, StudyPostStatus status) {
        return studyPostJpaRepository.findDeadlineReminderPosts(start, end, status);
    }

    @Override
    public Optional<StudyPost> findByIdWithAuthorForUpdate(UUID postId) {
        return studyPostJpaRepository.findByIdWithAuthorForUpdate(postId);
    }
}

File: infrastructure/src/main/kotlin/com/study/platform/domain/post/infrastructure/StudyPostRepositoryAdapter.kt
package com.study.platform.domain.post.infrastructure

import com.study.platform.domain.post.model.StudyPost
import com.study.platform.domain.post.model.StudyPostRepository
import com.study.platform.domain.post.model.StudyPostStatus
import org.springframework.stereotype.Repository
import java.time.LocalDateTime
import java.util.UUID

@Repository
class StudyPostRepositoryAdapter(
    private val studyPostJpaRepository: StudyPostJpaRepository
) : StudyPostRepository {

    override fun save(post: StudyPost): StudyPost =
        studyPostJpaRepository.saveAndFlush(post)

    override fun saveAll(posts: List<StudyPost>): List<StudyPost> =
        studyPostJpaRepository.saveAll(posts)

    override fun delete(post: StudyPost) {
        studyPostJpaRepository.delete(post)
    }

    override fun findById(id: UUID): StudyPost? =
        studyPostJpaRepository.findById(id).orElse(null)

    override fun findByIdWithAuthor(postId: UUID): StudyPost? =
        studyPostJpaRepository.findByIdWithAuthor(postId)

    override fun findExpiredPosts(now: LocalDateTime, status: StudyPostStatus): List<StudyPost> =
        studyPostJpaRepository.findExpiredPosts(now, status)

    override fun findDeadlineReminderPosts(start: LocalDateTime, end: LocalDateTime, status: StudyPostStatus): List<StudyPost> =
        studyPostJpaRepository.findDeadlineReminderPosts(start, end, status)

    override fun findByIdWithAuthorForUpdate(postId: UUID): StudyPost? =
        studyPostJpaRepository.findByIdWithAuthorForUpdate(postId)
}
